console/module/job/src/main/java/org/eclipse/kapua/app/console/module/job/server/GwtJobTargetServiceImpl.java

```diff
@@ -68,9 +68,12 @@
 
             // Converto to GWT entity
             for (JobTarget jt : jobTargetList.getItems()) {
-                gwtJobTargetList.add(KapuaGwtJobModelConverter.convertJobTarget(jt));
+                Device device = DEVICE_REGISTRY_SERVICE.find(jt.getScopeId(), jt.getJobTargetId());
+                if (device != null) {
+                    gwtJobTargetList.add(KapuaGwtJobModelConverter.convertJobTarget(jt));
+                    insertClientId(gwtJobTargetList);
```

Comment:
Why is this moved inside for-loop? It calls insertClientId multiple times with gwtJobTargetList.